bigbluebutton-html5/imports/ui/components/actions-bar/actions-dropdown/component.jsx

```diff
@@ -67,50 +76,69 @@ class ActionsDropdown extends Component {
     }
   }
 
-  handlePresentationClick() {
-    this.props.mountModal(<PresentationUploaderContainer />);
-  }
-
   getAvailableActions() {
     const {
       intl,
       handleShareScreen,
       handleUnshareScreen,
       isVideoBroadcasting,
+      isUserPresenter,
+      isUserModerator,
+      allowStartStopRecording,
+      isRecording,
+      record,
+      toggleRecording,
     } = this.props;
 
     return _.compact([
-      (<DropdownListItem
-        icon="presentation"
-        label={intl.formatMessage(intlMessages.presentationLabel)}
-        description={intl.formatMessage(intlMessages.presentationDesc)}
-        key={this.presentationItemId}
-        onClick={this.handlePresentationClick}
-      />),
-      (Meteor.settings.public.kurento.enableScreensharing ?
+      (isUserPresenter ?
+        <DropdownListItem
+          icon="presentation"
+          label={intl.formatMessage(intlMessages.presentationLabel)}
+          description={intl.formatMessage(intlMessages.presentationDesc)}
+          key={this.presentationItemId}
+          onClick={this.handlePresentationClick}
+        />
+        : null),
+      (Meteor.settings.public.kurento.enableScreensharing && isUserPresenter ?
         <DropdownListItem
           icon="desktop"
-          label={intl.formatMessage(isVideoBroadcasting ? intlMessages.stopDesktopShareLabel : intlMessages.desktopShareLabel)}
-          description={intl.formatMessage(isVideoBroadcasting ? intlMessages.stopDesktopShareDesc : intlMessages.desktopShareDesc)}
+          label={intl.formatMessage(isVideoBroadcasting ?
+            intlMessages.stopDesktopShareLabel : intlMessages.desktopShareLabel)}
+          description={intl.formatMessage(isVideoBroadcasting ?
+            intlMessages.stopDesktopShareDesc : intlMessages.desktopShareDesc)}
           key={this.videoItemId}
-          onClick={isVideoBroadcasting ? handleUnshareScreen : handleShareScreen }
+          onClick={isVideoBroadcasting ? handleUnshareScreen : handleShareScreen}
         />
-      : null),
+        : null),
+      (record && isUserModerator && allowStartStopRecording ?
+        <DropdownListItem
```

Comment:
I'd recommend the much simpler
`if (!allowStartStopRecording ....... ) return null;
<DropdownListItem .........>`